bindings_ffi/src/mls.rs

```diff
@@ -795,24 +946,40 @@ mod tests {
         )
         .await
         .unwrap();
+        register_client(&ffi_inbox_owner, &client).await;
+        return client;
+    }
 
-        let text_to_sign = client.text_to_sign().unwrap();
-        let signature = ffi_inbox_owner.sign(text_to_sign).unwrap();
+    #[tokio::test]
+    async fn get_inbox_id() {
+        let client = new_test_client().await;
+        let real_inbox_id = client.inbox_id();
 
-        client.register_identity(Some(signature)).await.unwrap();
-        client
+        let from_network = get_inbox_id_for_address(
+            Box::new(MockLogger {}),
+            xmtp_api_grpc::LOCALHOST_ADDRESS.to_string(),
+            false,
+            client.account_address.clone(),
+        )
+        .await
+        .unwrap()
+        .unwrap();
+
+        assert_eq!(real_inbox_id, from_network);
     }
 
     // Try a query on a test topic, and make sure we get a response
     #[tokio::test]
     async fn test_client_creation() {
         let client = new_test_client().await;
-        assert!(!client.account_address().is_empty());
+        assert!(!client.signature_request().is_none());
     }
 
     #[tokio::test]
+    #[ignore]
+    // This test needs to be updated to use the real address for the legacy signed private key
```

Comment:
I think this is the error case I'm running into since everything has to have legacy keys to support both V2 and V3.